File: Difficulty: Medium/Kth element in Matrix/kth-element-in-matrix.cpp
class Solution {
  public:
    int kthSmallest(vector<vector<int> >& matrix, int k) {
        int n = matrix.size();

        vector<int> result;                // store all matrix elements

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                result.push_back(matrix[i][j]);
            }
        }
        sort(result.begin(), result.end()); // sort ascending
        return result[k - 1];               // return k-th smallest (1-based)
    }
};

File: Difficulty: Medium/Kth element in Matrix/kth-element-in-matrix.java
class Solution {
    public int kthSmallest(int[][] matrix, int k) {
        int n = matrix.length;

        List<Integer> result = new ArrayList<>(); // store all matrix elements

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                result.add(matrix[i][j]);
            }
        }
        result.sort(null);        // sort ascending
        return result.get(k - 1); // return k-th smallest (1-based)
    }
}